src/components/graph-view.js

```diff
@@ -1377,19 +1378,21 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
     );
   };
 
-  asyncRenderNode(
-    node: INode,
-    renderEdges: boolean = true,
-    onRendered: () => void = () => {}
-  ) {
+  asyncRenderNode(node: INode, renderEdges: boolean = true) {
     const nodeKey = this.props.nodeKey;
-    const timeoutId = `nodes-${node[nodeKey]}`;
+    const nodeRenderId = `nodes-${node[nodeKey]}`;
 
-    cancelAnimationFrame(this.nodeTimeouts[timeoutId]);
-    this.nodeTimeouts[timeoutId] = requestAnimationFrame(() => {
+    cancelAnimationFrame(this.nodeTimeouts[nodeRenderId]);
+    this.nodeTimeouts[nodeRenderId] = requestAnimationFrame(() => {
       this.syncRenderNode(node, renderEdges);
-      onRendered();
+
+      if (this.onNodeRenderCallbacks[nodeRenderId]) {
+        this.onNodeRenderCallbacks[nodeRenderId]();
+        delete this.onNodeRenderCallbacks[nodeRenderId];
```

Comment:
why delete

Reply:
I delete the callback here because I don't want subsequent callers to also trigger the same callback -- it should only be triggered once.